src/vdom/component.js

```diff
@@ -158,15 +175,8 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		}
 
 		component.base = base;
-		if (base && !isChild) {
-			let componentRef = component,
-				t = component;
-			while ((t=t._parentComponent)) {
-				(componentRef = t).base = base;
-			}
-			base._component = componentRef;
-			base._componentConstructor = componentRef.constructor;
-		}
+		base._component = component;
```

Comment:
Removing the conditional & crawl here seems like it would break component composition. When re-rendering a component that is a child of another component, it will update the `_component` property in the DOM do point to itself, which is then incorrect.

Reply:
I removed these changes in https://github.com/developit/preact/pull/886 and verified the tests still pass. Wasn't sure how/if I could amend this PR directly.